lxqt-config-appearance/main.cpp

```diff
@@ -104,21 +107,27 @@ int main (int argc, char **argv)
     SelectWnd* cursorPage = new SelectWnd(sessionSettings, dialog);
     cursorPage->setCurrent();
     dialog->addPage(cursorPage, QObject::tr("Cursor"), QStringList() << QStringLiteral("input-mouse") << QStringLiteral("preferences-desktop"));
-    QObject::connect(cursorPage, &SelectWnd::settingsChanged, dialog, [dialog] {
-        dialog->enableButton(QDialogButtonBox::Apply, true);
-    });
+    std::shared_ptr<bool> cursorChanged = std::make_shared<bool>(false);
+    QObject::connect(cursorPage, &SelectWnd::settingsChanged, dialog, [dialog, cursorChanged] {
+            dialog->enableButton(QDialogButtonBox::Apply, true);
+            *cursorChanged = true;
+            });
```

Comment:
I'm not sure whether it's necessary to show a message but, in case it is:

Instead of this, wouldn't it be better if `SelectWnd` had a small method to tell whether the cursor was changed or not? I think a private boolean and a public method that gets it would be enough.

Reply:
OK, I tested under X11 (our main goal) and the cursor was changed on the fly. The message isn't needed.

If it's needed under Wayland, please limit it to Wayland by adding the above-mentioned method to `SelectWnd`. However, I think it's better not to show any message under Wayland either.

Reply:
> OK, I tested under X11 (our main target) and the cursor was changed on the fly. The message isn't needed.

Open a LXQt session with openbox as window manager. Then change the cursor style. Finally, put your cursor over an openbox title bar. Openbox keeps the previous cursor theme.
I have installed mousepad in my computer to test GTK3 themes, and mousepad doesn't change cursor theme.

I have just found this: If you see on top of SelectWnd, there is this text: "LXQt needs restart after this change". I will remove the dialog.

Thanks for your review.

Reply:
> Openbox keeps the previous cursor theme.

I tested with KWin; I don't use Openbox. But you're right.